homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -71,10 +81,65 @@ def get_platforms(config_entry):
         for air_monitor_model in MODELS_AIR_MONITOR:
             if model.startswith(air_monitor_model):
                 return AIR_MONITOR_PLATFORMS
-
+    _LOGGER.error(
+        "Unsupported device found! Please create an issue at "
+        "https://github.com/syssi/xiaomi_airpurifier/issues "
+        "and provide the following data: %s",
+        model,
+    )
     return []
 
 
+async def async_create_miio_device_and_coordinator(
+    hass: core.HomeAssistant, entry: config_entries.ConfigEntry
+):
+    """Set up a data coordinator and one miio device to service multiple entities."""
+    model = entry.data[CONF_MODEL]
+    host = entry.data[CONF_HOST]
+    token = entry.data[CONF_TOKEN]
+    name = entry.title
+    device = None
+    if model not in MODELS_HUMIDIFIER:
+        return
+    if model in MODELS_HUMIDIFIER_MIOT:
+        device = AirHumidifierMiot(host, token)
+    else:
+        device = AirHumidifier(host, token, model=model)
+
+    # removing fan platform entity for humidifiers
+    entity_registry = await en.async_get_registry(hass)
```

Comment:
```suggestion
    entity_registry = er.async_get(hass)
```

Reply:
Tried this out. Think we should not remove await here:
```log
/usr/local/lib/python3.8/logging/handlers.py:1482: RuntimeWarning: coroutine 'async_get_registry' was never awaited
Coroutine created at (most recent call last)
  File "/workspaces/core/homeassistant/runner.py", line 118, in run
    return asyncio.run(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 603, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 570, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 1851, in _run_once
    handle._run()
  File "/usr/local/lib/python3.8/asyncio/events.py", line 95, in _run
    self._context.run(self._callback, *self._args)
  File "/workspaces/core/homeassistant/config_entries.py", line 293, in async_setup
    result = await component.async_setup_entry(hass, self)  # type: ignore
  File "/workspaces/core/homeassistant/components/xiaomi_miio/__init__.py", line 57, in async_setup_entry
    or await async_setup_device_entry(hass, entry)
  File "/workspaces/core/homeassistant/components/xiaomi_miio/__init__.py", line 223, in async_setup_device_entry
    await async_create_miio_device_and_coordinator(hass, entry)
  File "/workspaces/core/homeassistant/components/xiaomi_miio/__init__.py", line 110, in async_create_miio_device_and_coordinator
    entity_registry = er.async_get_registry(hass)
  record = self.dequeue(True)
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```

Reply:
`async_get` isn't async https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity_registry.py#L594 so `entity_registry = er.async_get(hass)` is correct.

I think you used `async_get_registry` without `await`.

Reply:
My bad. I only removed await and renamed en to er, but i forgot to remove the `_registry` part.
I think the suggested changes are fine!